velero-plugins/pod/restore.go

```diff
@@ -113,9 +113,37 @@ func (p *RestorePlugin) Execute(input *velero.RestoreItemActionExecuteInput) (*v
 	destStagePodImage := input.Restore.Annotations[common.StagePodImageAnnotation]
 	if len(pod.Labels[common.IncludedInStageBackupLabel]) > 0 && len(destStagePodImage) > 0 {
 		p.Log.Infof("[pod-restore] swapping stage pod images for pod %s", pod.Name)
+		pvcVolumes := []corev1API.Volume{}
+		exclude := []string{}
+		for _, volume := range pod.Spec.Volumes {
+			if volume.PersistentVolumeClaim == nil {
+				exclude = append(exclude, volume.Name)
+				continue
+			}
+			pvcVolumes = append(pvcVolumes, volume)
```

Comment:
We also need to filter this to exclude "move" and "snapshot copy" volumes from the pod, since these are excluded from stage restore, and the pod won't come up if it's missing volumes. What's not obvious is how to do that here. 
@dymurray do you have any insight here? When creating stage pods in the controller, we can use the MigPlan selection values. We don't have access to the PVC annotations since those are in the backup, not in the cluster. Do we need to annotate application pods with a list of PVC exclusions? Basically when we add the stage labels to application pods on backup, in GetApplicationPodsWithStageLabels, when we find a PVC that should be excluded from the stage pod because it's a move or snapshot copy, we'll add its name to a PVC exclude list annotation, and the plugin here could make use of that. Is there a cleaner way to do this?

Reply:
+1